src/transformers/modeling_t5.py

```diff
@@ -591,19 +591,15 @@ def _init_weights(self, module):
                 module.relative_attention_bias.weight.data.normal_(mean=0.0, std=factor * ((d_model) ** -0.5))
 
     def _shift_right(self, input_ids):
-        decoder_start_token_id = self.config.decoder_start_token_id
         pad_token_id = self.config.pad_token_id
 
-        assert (
-            decoder_start_token_id is not None
-        ), "self.model.config.decoder_start_token_id has to be defined. In T5 it is usually set to the pad_token_id. See T5 docs for more information"
+        assert pad_token_id is not None, "self.model.config.pad_token_id has to be defined."
 
         # shift inputs to the right
         shifted_input_ids = input_ids.new_zeros(input_ids.shape)
         shifted_input_ids[..., 1:] = input_ids[..., :-1].clone()
-        shifted_input_ids[..., 0] = decoder_start_token_id
+        shifted_input_ids[..., 0] = pad_token_id
```

Comment:
same as below. T5 should not mix `pad_token_id` with `decoder_start_token_id` with `bos_token_id` (which does not exist). If `bos_token_id` would exist, I'd be fine to use it here. `pad_token_id` is confusing for the user because logically it should be the `decoder_start_token_id` (even though they are equal in T5).

Reply:
rolled back decoder_start_token_id changes